pkg/cluster/spec/spec.go

```diff
@@ -495,9 +501,32 @@ func (s *Specification) CountDir(targetHost, dirPrefix string) int {
 	return count
 }
 
+func (s *Specification) validateTiSparkSpec() error {
+	if s.TiSparkMasters == nil && s.TiSparkWorkers == nil {
+		return nil
+	}
+	if len(s.TiSparkMasters) == 0 {
+		if s.TiSparkWorkers == nil || len(s.TiSparkWorkers) == 0 {
+			return nil
+		}
+		return ErrNoTiSparkMaster
+	}
+
+	// We only support 1 spark master at present
+	if s.TiSparkMasters != nil && len(s.TiSparkMasters) > 1 {
+		return ErrMultipleTiSparkMaster
+	}
+
+	return nil
```

Comment:
This function can be simplified if we use `len` function instead of check the `nil` value.